### DISCUSSION TITLE
submodule 도입에 대해

### DISCUSSION THREAD

**Discussion by Choi-JJunho** [woowacourse-teams/2023-pium#324] [개발 논의]
# submodule 도입

> 다음 논의로부터 이어집니다.
- #187 

안녕하십니까 친애하는 피움 팀원여러분
저희는 현재 빌드 서버 내부에 properties / env 파일들을 관리하고있습니다.

이로인해 다음과 같은 문제가 발생하고 있습니다.
- 캠퍼스 외부에서 properties / env 파일 관리 불가
-  properties 설정을 파악하기가 어려움 (vim, cat 명령어로 확인)
- 분업화 진행 간 properties / env 파일 변경에 대한 공유가 어려움

이에 다음과 같은 기술의 도입에 대해 건의드립니다.

## submodule

git submodule 를 이용하여 프로젝트 내부에 properties 파일을 별도로 받아오는 방식을 생각해볼 수 있습니다.
대략적인 흐름도는 다음과 같습니다.

![image](https://github.com/woowacourse-teams/2023-pium/assets/49794401/56661283-8f6b-437a-9d1d-2d61a27b04cc)
[이미지 출처](https://bionicape.com/unreal-plugins-as-git-submodules/)

예시는 python 파일에 대한 내용이나 저희가 적용하게 된다면 front, back 설정파일부분에 properties / env 파일을 적용하는 방식일 될 것 같습니다.

만약 도입하게 된다면 properties / env 파일은 [피움 Organization](https://github.com/pium-official)의 private Repository에서 관리하게 될 것 같습니다.

submodule 도입 과정에 대해 여러분들의 의견이 궁금합니다!

## 참고해보면 좋을만한 자료들
- [Git - 서브모듈](https://git-scm.com/book/ko/v2/Git-%EB%8F%84%EA%B5%AC-%EC%84%9C%EB%B8%8C%EB%AA%A8%EB%93%88)
- [Tecoble - git submodule로 중요한 정보 관리하기](https://tecoble.techcourse.co.kr/post/2021-07-31-git-submodule/)

**Comment by Kim0914** [comment]
`캠퍼스 외부에서 properties / env 파일 관리 불가`

이 부분이 가장 공감이 되는 것 같네요 ㅎㅎ

서브모듈로 관리하는 방안 좋아요 !

**Comment by WaiNaat** [comment]
![좋아요](https://github.com/woowacourse-teams/2023-pium/assets/77872742/8ab320d1-532b-4188-80e6-82a91fe893eb)

**Comment by yeonkkk** [comment]
properties 파일 관리하기 더 좋겠네요 !! 
저도 서브모듈로 관리하는 거 좋습니다!ㅎㅎ

**Comment by Choi-JJunho** [comment]
다음 PR에서 작업을 진행했습니다
- #356
